tiledb/common/random/random_label.h

```diff
@@ -33,10 +33,91 @@
 #ifndef TILEDB_HELPERS_H
 #define TILEDB_HELPERS_H
 
+#include <iomanip>
+#include <mutex>
+#include <sstream>
 #include <string>
 
+#include "tiledb/common/exception/exception.h"
+#include "tiledb/common/random/prng.h"
+#include "tiledb/sm/misc/tdb_time.h"
+
 namespace tiledb::common {
 
+class RandomLabelException : public StatusException {
+ public:
+  explicit RandomLabelException(const std::string& message)
+      : StatusException("RandomLabel", message) {
+  }
+};
+
+class RandomLabelGenerator {
+ public:
+  /* ********************************* */
+  /*     CONSTRUCTORS & DESTRUCTORS    */
+  /* ********************************* */
+  DISABLE_COPY_AND_COPY_ASSIGN(RandomLabelGenerator);
+  DISABLE_MOVE_AND_MOVE_ASSIGN(RandomLabelGenerator);
+
+  /** Default destructor. */
+  ~RandomLabelGenerator() = default;
+
+ protected:
+  /** Protected constructor, abstracted by public-facing accessor. */
+  RandomLabelGenerator();
+
+  /* ********************************* */
+  /*                API                */
+  /* ********************************* */
+  /** Generate a random label. */
+  std::string generate() {
+    std::lock_guard<std::mutex> lock(mtx_);
+    PRNG& prng = PRNG::get();
```

Comment:
Could you move the `std::lock_guard` to after the `PRNG::get()`? I can't immediately convince myself there's not a dead lock condition here so better safe than sorry. I'm pretty sure `prng()` locks as well, but I'm fairly certain that's fine once both of these global singletons are isntantiated.